raphtory/src/db/view_api/internal/exploded_edge_ops.rs

```diff
@@ -0,0 +1,100 @@
+use crate::core::edge_ref::EdgeRef;
+use crate::core::vertex_ref::LocalVertexRef;
+use crate::core::Direction;
+use crate::db::view_api::internal::time_semantics::TimeSemantics;
+use crate::db::view_api::internal::GraphOps;
+use crate::db::view_api::BoxedIter;
+use std::ops::Range;
+
+/// Additional methods for returning exploded edge data that are automatically implemented
+pub trait ExplodedEdgeOps {
+    /// Returns an iterator over the exploded edges connected to a given vertex in a given direction.
+    ///
+    /// # Arguments
+    ///
+    /// * `v` - A reference to the vertex for which the edges are being queried.
+    /// * `d` - The direction in which to search for edges.
+    ///
+    /// # Returns
+    ///
+    /// Box<dyn Iterator<Item = EdgeRef> + Send> -  A boxed iterator that yields references to
+    /// the edges connected to the vertex.
+    fn vertex_edges_t(
```

Comment:
Can we rename all _t functions to something like _exploded to make them a little clearer.

Reply:
Yes, let's do that later